core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java

```diff
@@ -176,12 +176,12 @@ protected NodeResponse nodeOperation(NodeRequest request, Task task) {
             this.request = request;
             this.listener = listener;
             ClusterState clusterState = clusterService.state();
-            String[] nodesIds = resolveNodes(request, clusterState);
-            this.nodesIds = filterNodeIds(clusterState.nodes(), nodesIds);
-            ImmutableOpenMap<String, DiscoveryNode> nodes = clusterState.nodes().getNodes();
+            String[] nodesIdCandidates = resolveNodes(request, clusterState);
+            nodesIds = filterNodeIds(clusterState.nodes(), nodesIdCandidates);
+            ImmutableOpenMap<String, DiscoveryNode> nodesCandidates = clusterState.nodes().getNodes();
```

Comment:
We should also get rid of this local variable (The map is not needed as `DiscoveryNodes` has a method `get()`).

We could write `this.nodes[i] = clusterState.nodes().get(nodesIds[i]);` a few lines below.

If `clusterState.nodes()` is too verbose, we can extract that one into a local variable.